go/saltpack/example_test.go

```diff
@@ -0,0 +1,114 @@
+package saltpack_test
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/keybase/client/go/saltpack"
+	"github.com/keybase/client/go/saltpack/basic"
+	"io"
+	"os"
+)
+
+func ExampleEncryptArmor62Seal() {
+
+	var err error
+
+	// Make a new Keyring, initialized to be empty
+	basicKeyring := basic.NewKeyring()
+	var keyring saltpack.Keyring
+	keyring = basicKeyring
+
+	// The test message
+	msg := []byte("The Magic Words are Squeamish Ossifrage")
+
+	// Make a secret key for the sender
+	var sender saltpack.BoxSecretKey
+	sender, err = basicKeyring.GenerateBoxKey()
+	if err != nil {
+		return
+	}
+
+	// And one for the receiver
+	var receiver saltpack.BoxSecretKey
+	receiver, err = basicKeyring.GenerateBoxKey()
+	if err != nil {
+		return
+	}
+
+	// AllReceivers can contain more receivers (like the sender)
+	// but for now, just the one.
+	var ciphertext string
+	allReceivers := []saltpack.BoxPublicKey{receiver.GetPublicKey()}
+	ciphertext, err = saltpack.EncryptArmor62Seal(msg, sender, allReceivers, "")
+	if err != nil {
+		return
+	}
+
+	// The decrypted message
+	var msg2 []byte
+	_, msg2, _, err = saltpack.Dearmor62DecryptOpen(ciphertext, keyring)
```

Comment:
It would certainly work.  I downcast to the interface though to be clearer about what the types are (for the purposes of the example).  I can obviously get rid of it though!